### PR TITLE
AP_HAL_F4Light: translated comments,

### PR DESCRIPTION
**Pull request by night-ghost**
fixes for scripts, parameters divided to common and board specific, parameter to reboot into DFU mode

### CODE REVIEW

**Comment by khancyr** [comment]
Looks good.
Only the PR title is wrong but not a big deal. The fix are limited to F4light hal, and generally look good, so safe to merge if they have been tested !

**Comment by night-ghost** [comment]
last changes in one PR but separated by commits. Switching to using bootloader broke "reboot to DFU" feature, that together with broken "reboot to bootloader" in bootloader itself becomes to impossibility to flash board without BOOT button. Whch is very hard if FC board mounted very deep in aircraft.
 I got a lots of bugreports, this PR and next one fixes both issues.

File: libraries/AP_Param_Helper/AP_Param_Helper.cpp [hunk 1]
```diff
@@ -5,9 +5,11 @@
 #if defined(HAL_NEEDS_PARAM_HELPER)
 
 const AP_Param::GroupInfo AP_Param_Helper::var_info[] = {
+    F4LIGHT_HAL_VARINFO
```
Comment (OXINARF) [inline libraries/AP_Param_Helper/AP_Param_Helper.cpp]: I know that no other HAL uses this now, but, as it is a generic library, we should put a macro around this:

```
#if CONFIG_HAL_BOARD == HAL_BOARD_F4LIGHT
    F4LIGHT_HAL_VARINFO
#endif
```

File: libraries/AP_Param_Helper/AP_Param_Helper.h [hunk 2]
```diff
@@ -16,7 +16,10 @@ class AP_Param_Helper
 
     static const AP_Param::GroupInfo var_info[];
 
-// only if defined
+// common HAL params
+    F4LIGHT_HAL_PARAMS
```
Comment (OXINARF) [inline libraries/AP_Param_Helper/AP_Param_Helper.h]: Same here.

File: libraries/AP_Param_Helper/AP_Param_Helper.cpp (hunk 1 above)
Comment (night-ghost) [reply libraries/AP_Param_Helper/AP_Param_Helper.cpp]: this is not good approach, it should be selected by feature, not by name. Added #if  defined() so any board could to use parameters this way. As it did with HAL_REQUIRES_PARAM_HELPER

Comment (OXINARF) [reply libraries/AP_Param_Helper/AP_Param_Helper.cpp]: That's fine.

**Comment by OXINARF** [comment]
This isn't passing Travis so I'm holding merging this yet. I believe this is an issue with Travis and not with this PR and I'll be monitoring Travis and re-running the failed job.

**Comment by night-ghost** [comment]
please see log - just timeout on tests

**Comment by OXINARF** [comment]
It's not a timeout, it's failing to run the SITL quadplane test. It starts crashed on the ground, which I believe is a timing issue with Travis, rather than a real problem. Travis is having degraded performance so we'll need to wait that it gets fixed.

**Comment by night-ghost** [comment]
I can do forced push if it needed

**Comment by OXINARF** [comment]
No need. As I expected this has now passed Travis. Merged, thanks!

**Pull request merged**